File: week-1/self-dividing-numbers/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []
        for num in range(left, right+1):
            temp = num
            flag = False
            while temp:
                digit = temp%10
                temp //= 10
                if digit == 0 or num % digit != 0:
                    flag = True
                    break
                    
            if not flag:
                res.append(num)
        
        return res

File: week-1/self-dividing-numbers/self-dividing-numbers.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var selfDividingNumbers = function(left, right) {
    let res = [];
    for (let i=left; i<=right; i++) {
        let temp = i
        let flag = false;
        while (temp) {
            digit = temp%10
            temp = Math.floor(temp/10)
            if (digit === 0 || i % digit !== 0) {
                flag = true;
                break;
            }
        }
        if (!flag) {
            res.push(i)
        }
    }
    
    return res;
};
